### PR TITLE
Adding AdsDataHub basic workers

### PR DESCRIPTION
**Pull request by gbarbier-google**
This commit adds one abstract worker, a query launcher and a waiter
to enable running pre-existing AdsDataHub queries from CRMint

As discussed with @dulacp

### CODE REVIEW

File: backends/core/workers.py [hunk 1]
```diff
@@ -1842,3 +1843,65 @@ def _set_target_column(self, client, dataset_resource_name, target_column_spec):
...
+              query_id=self._params['query_id'],
+              customer_id=self._params['customer_id']),
+          body=json.loads(self._params['payload'])
+    ).execute()
+
+    self.log_info("New operation %s created", operation['name'])
+
+    self._enqueue('AdsDataHubQueryWaiter',
+                    {'operation_name': operation['name']}, 60)
+
```
Comment (dulacp) [inline backends/core/workers.py:1885]: Same here for spacing between classes.

File: backends/core/workers.py [hunk 2]
```diff
@@ -1842,3 +1843,65 @@ def _set_target_column(self, client, dataset_resource_name, target_column_spec):
...
+  def _execute(self):
+    self._get_adh_client()
+
+    operation = self._adh_client.customers().analysisQueries().start(
+        name='customers/{customer_id}/analysisQueries/{query_id}'
+          .format(
+              query_id=self._params['query_id'],
+              customer_id=self._params['customer_id']),
+          body=json.loads(self._params['payload'])
```
Comment (dulacp) [inline backends/core/workers.py:1878]: This should be aligned with `name=` I believe

File: backends/core/workers.py [hunk 3]
```diff
@@ -1842,3 +1843,65 @@ def _set_target_column(self, client, dataset_resource_name, target_column_spec):
...
+    ).execute()
+
+    self.log_info("New operation %s created", operation['name'])
```
Comment (dulacp) [inline backends/core/workers.py:1881]: Prefer using only one type of quote (single or double) in the document.

File: backends/core/workers.py [hunk 4]
```diff
@@ -1842,3 +1843,65 @@ def _set_target_column(self, client, dataset_resource_name, target_column_spec):
...
+
+  def _execute(self):
+    self._get_adh_client()
+
+    operation = self._adh_client.operations().get(
+        name=self._params['operation_name']).execute()
+
+    # API is not behaving as expected, i.e. done is not added to response
+    # before the execution is finished
+    if 'done' not in operation or operation['done'] != True:
```
Comment (dulacp) [inline backends/core/workers.py:1901]: You could simplify that in python like this

```python
if not operation.get('done', False):
  ...
```

With `False` being the default value used if 'done' is not in operation.

**Review by dulacp — CHANGES_REQUESTED** [review]
A few formatting comments.

File: backends/core/workers.py (hunk 2 above)
Comment (gbarbier-google) [reply backends/core/workers.py]: I aligned `body=` with `name=` but kept the other indented to the right - hope it makes sense for you too

**Review by dulacp — APPROVED** [review]
LGTM

**Pull request merged**